commander/src/bootstrapping/commands/transaction/sign.ts

```diff
@@ -253,7 +197,10 @@ export abstract class SignCommand extends Command {
 			description: 'Optional publicKey string in hex format.',
 		}),
 		'chain-id': flagsWithParser.chainID,
-		'sender-public-key': flagsWithParser.senderPublicKey,
+		'sender-public-key': {
+			...flagsWithParser.senderPublicKey,
+			relationships: [{ type: 'some', flags: ['mandatory-keys', 'optional-keys'] } as Relationship],
+		},
```

Comment:
looking at the behavior, i think now `sender-public-key` is not used anywhere in the code other than condition for offline signing.
That condition also can be replaced with mandatory-keys/optional-keys check?

Reply:
True. This command only signs an existing transaction. It means that the provided transaction should already have `senderPublicKey` property populated as part of `transaction:create` command.